src/main/java/car_rental/api/user/UserAppDTO.java

```diff
@@ -0,0 +1,141 @@
+package car_rental.api.user;
+
+import car_rental.api.utils.PasswordMatcher;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Objects;
+
+@PasswordMatcher()
+public class UserAppDTO {
+
+    @JsonProperty("username")
+    @NotEmpty(message = "This can not be empty")
+    @NotNull(message = "This can not be null")
+    @Size(min = 5, message = "Try one with at least 5 characters")
+    private String username;
+
+    @JsonProperty("email")
+    @NotEmpty(message = "This can not be empty")
+    @NotNull(message = "This can not be null")
+    @Email
+    private String email;
+
+    @JsonProperty("password")
+    @NotEmpty(message = "This can not be empty")
+    @NotNull(message = "This can not be null")
+    @Size(min = 6, message = "Try one with at least 6 characters")
+    private String password;
+
+    @JsonProperty("confirmPassword")
+    @NotEmpty(message = "This can not be empty")
+    @NotNull(message = "This can not be null")
+    private String confirmPassword;
+
+    public String getUsername() {
+        return username;
+    }
+
+    public UserAppDTO() {
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getConfirmPassword() {
+        return confirmPassword;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public void setConfirmPassword(String confirmPassword) {
+        this.confirmPassword = confirmPassword;
+    }
+
+
+
+    @Override
+    public String toString() {
+        return "UserAppDTO{" +
+                "username='" + username + '\'' +
+                ", email='" + email + '\'' +
+                ", password='" + password + '\'' +
+                ", confirmPassword='" + confirmPassword + '\'' +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
To jest wygenerowany kod czy sam to pisałeś?

Reply:
generowany w inteliJJ